### PR TITLE
[WIP] Exporter Cannon

### PR DESCRIPTION
**Pull request by Core0verload**
This PR adds x-porter cannon, weapon of a true capitalist, to the cargo. It comes in a 20 000 credits contraband crate.

This cannon sells any object it's fired at. Selling affects the contents, so if you fired it at backpack full of plasma, the backpack would remain on the floor, but plasma inside would be sold.

When emagged, the cannon will try to sell the gear of any mob it hits. Similar to acid mix spray, but less lethal and more profitable. Perfect for dealing with red commie secs attacking your capitalistic heaven.

You can aim the cannon at mouth on "help" intent to perform a suicide/execution, just like with any other gun. Doing so will sell target's internal organs on the black market.

TODO:
- [x] Remove debug messages.
- [x] Add logging.
- [x] Implement emagged gear selling.
- [x] Implement execution organs selling.
- [x] Recharge time dependent on amount of credits recieved from the sale.
- [x] New exports for getting rid of things like unscrewed windows or girders.
- [x] Add the gun to cargo contraband crates.
- [ ] "Money spray" effect on successful sale.
- [ ] TEST EVERYTHING!

### CODE REVIEW

**Comment by Bawhoppen** [comment]
why a contraband crate

**Comment by Core0verload** [comment]
Too fun for a regular one. And I want it to be contraband so it can perform contraband organ sales by default.

**Comment by MMMiracles** [comment]
the mental image of literally selling the clothes off of security's back as they chase you is too good to pass up

i don't know if that would be better or worse than just dying as an officer, which makes it even better.

**Comment by Yiu113** [comment]
The most important question, of course, is if people with organs from aliens will sell for more. How much more will the Abductor organs sell for, compared to a normal human?

File: code/modules/projectiles/guns/energy/special.dm [hunk 1]
```diff
@@ -302,4 +302,23 @@
 	ammo_type = list(/obj/item/ammo_casing/energy/instakill/blue)
 
 /obj/item/weapon/gun/energy/laser/instakill/emp_act() //implying you could stop the instagib
-	return
\ No newline at end of file
+	return
+
+
+/obj/item/weapon/gun/energy/xporter
+	name = "x-porter cannon"
```
Comment (Iamgoofball) [inline code/modules/projectiles/guns/energy/special.dm]: rename to sell-a-porter

File: code/modules/projectiles/projectile/special.dm [hunk 2]
```diff
@@ -245,4 +245,77 @@
...
+	var/obj/item/weapon/gun/energy/xporter/gun
+	var/obj/docking_port/mobile/supply/supply
+	var/selling = 0
+
+/obj/item/projectile/energy/sell/on_hit(atom/target, blocked = 0)
+	. = ..()
+
+	if(iscarbon(target) && blocked != 100)
+		if(damage >= 5) // Execution? Sell victim's organs!
+			world << "exporter execution"
```
Comment (Iamgoofball) [inline code/modules/projectiles/projectile/special.dm]: world <<

**Comment by Shadowlight213** [comment]
Yes. So much yes.
I mentioned to goof how we needed the ability to sell the shirt off of people's backs.

File: code/modules/projectiles/projectile/special.dm (hunk 2 above)
Comment (Jalleo) [inline code/modules/projectiles/projectile/special.dm]: Yeah that must be a debug thing

File: code/modules/projectiles/guns/energy/special.dm [hunk 3]
```diff
@@ -302,4 +302,23 @@
...
-	return
\ No newline at end of file
+	return
+
+
+/obj/item/weapon/gun/energy/xporter
+	name = "x-porter cannon"
+	desc = "An industrial-grade teleporter connected to export scanner. Weapon of a true capitalist."
+	icon_state = "export_cannon"
+	ammo_type = list(/obj/item/ammo_casing/energy/xport)
```
Comment (Iamgoofball) [inline code/modules/projectiles/guns/energy/special.dm]: gonna need a shotgun sell-a-porter variant

**Comment by Armhulen** [comment]
>not making it sell every organ in their body
can't tell if this is fun removal just because of the fact that you can't do that

File: code/modules/projectiles/guns/energy/special.dm (hunk 1 above)
Comment (PKPenguin321) [inline code/modules/projectiles/guns/energy/special.dm]: Jew ray

**Comment by AdamElTablawy** [comment]
i can't wait for antag cargo techs to emag this and sell the fucking HOS as he tries to taze you, leaving him naked and embarassed, and sell the brig doors, and sell the way into the armory, and sell the entire armory

like it sounds like it'd be funny for a meme but i can see this becoming really unbalanced and a huge grief tool if some cockhead ran into medbay and sold all the medkits and the supplies and the IV and the blood and the tools and the beakers, and slipped a miner and sold the locker full of minerals he was dragging, and everything coming out of science

**Comment by optimumtact** [comment]
This is the one thing I hated about goof cargo, it just feels too memey

:-1:

**Comment by KorPhaeron** [comment]
>This is the one thing I hated about goof cargo, it just feels too memey

**Comment by Eaglendia** [comment]
I mean it costs 20k which is 6k more than the bundle containing >20 TC of traitor items.

The sellaporter was fucking hilarious and I'd love to see it back in a form like this, where it's a contraband item as opposed to just having two lying around at roundstart. Nothing wrong with at least testing it if it gets done.

**Comment by KorPhaeron** [comment]
Stale

**Comment by PKPenguin321** [comment]
:(

**Comment by Core0verload** [comment]
I'm still working on it, I just don't commit until it's ready. Re-open it.

**Comment by optimumtact** [comment]
I wish to delete the meme cannon

**Comment by PKPenguin321** [comment]
no you are not allowed

**Comment by Core0verload** [comment]
Admin logging is in. The only thing that isn't ready yet is a fancy "on sell" effect and sound.
>I wish to delete the meme cannon

Oh, what happened to our Memelord of Dank?

**Comment by Core0verload** [comment]
Can someone find a cash register ka-ching sound? I want my exporter to sound like capitalism.

**Comment by Iamgoofball** [comment]
check my old PR for the port-a-seller, I had a pretty capitalist sound for that

**Comment by Iamgoofball** [comment]
https://raw.githubusercontent.com/Iamgoofball/-tg-station/15d290add12f569f4fb2b1c9a1a4abcbc50e3ebd/sound/effects/import_sound.wav

**Comment by swankcookie** [comment]
I'm just imagining a loony tunes scenario where an officer is chasing someone, and suddenly, still running, he's completely naked.

**Comment by optimumtact** [comment]
Stale?

**Comment by KorPhaeron** [comment]
Yes

**Comment by Core0verload** [comment]
one day i will finish it

not today

**Comment by swankcookie** [comment]
please finish this... please

**Pull request closed without merging**